File: 4-sorting/sort-random-array.js
let arr = [];

console.time(`Array generation`);

for (let i = 0; i < 1e7; i++) {
	arr[i] = Math.floor(Math.random() * 1e6);
}

console.timeEnd(`Array generation`);

console.time(`Array sorting`);

arr.sort((a, b) => a - b);

console.timeEnd(`Array sorting`);

File: 4-sorting/sort-random-array.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <cstdio>
#include <vector>
#include <chrono>
#include <algorithm>

using namespace std;

int main() {
	const int size = 1e7;

	srand(time(nullptr));

	auto start = chrono::high_resolution_clock::now();

	vector<int> arr(size);

	for (int i = 0; i < size; i++) {
		arr[i] = rand() % (int)1e6;
	}

	auto end = chrono::high_resolution_clock::now();
	auto duration = chrono::duration_cast<chrono::milliseconds>(end - start).count();

	cout << "Array generated in " << duration << "ms." << endl;

	start = chrono::high_resolution_clock::now();

	sort(arr.begin(), arr.end());

	end = chrono::high_resolution_clock::now();
	duration = chrono::duration_cast<chrono::milliseconds>(end - start).count();

	cout << "Array sorted in " << duration << "ms." << endl;

	return EXIT_SUCCESS;
}